packages/react-native/ReactCommon/react/performance/timeline/PerformanceEntryKeyedBuffer.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "PerformanceEntryKeyedBuffer.h"
+#include <string>
+
+namespace facebook::react {
+
+void PerformanceEntryKeyedBuffer::add(const PerformanceEntry& entry) {
+  auto& list = entryMap_.at(entry.name);
+  list.push_back(entry);
+  totalEntryCount_ += 1;
+}
+
+void PerformanceEntryKeyedBuffer::getEntries(
+    std::optional<std::string_view> name,
+    std::vector<PerformanceEntry>& target) const {
+  if (name.has_value()) {
+    std::string nameStr{name.value()};
+
+    if (auto node = entryMap_.find(nameStr); node != entryMap_.end()) {
+      target.insert(target.end(), node->second.begin(), node->second.end());
+    }
+  } else {
+    getEntries(target);
+  }
+}
+
+void PerformanceEntryKeyedBuffer::clear() {
+  entryMap_.clear();
+  totalEntryCount_ = 0;
+}
+
+void PerformanceEntryKeyedBuffer::clear(std::string_view nameView) {
+  std::string name{nameView};
+
+  if (auto node = entryMap_.find(name); node != entryMap_.end()) {
+    totalEntryCount_ -= node->second.size();
+    node->second.clear();
```

Comment:
We should probably clear the whole entry and not just the vector it contains, so we reduce memory usage.